File: Presentation/src/app/races/shared/models/team-result-view-model.ts
import { StageResultViewModel } from '.';
import { TeamCategory } from './team-category';

export class TeamResultViewModel {
    public teamNumber: number;
    public teamName: string;
    public category: TeamCategory;
    public endTime: Date;
    public raceDuration: string;
    public totalValue: number;
    public numberOfPoints: number;
    public numberOfStages: number;
    public stageResults: StageResultViewModel[];
}

File: RaceVenturaAPI/ViewModels/Races/TeamResultViewModel.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace RaceVenturaAPI.ViewModels.Races
{
    public class TeamResultViewModel
    {
        public int TeamNumber { get; set; }
        public string TeamName { get; set; }
        public DateTime? EndTime { get; set; }

        [JsonConverter(typeof(JsonTimeSpanConverter))]
        public TimeSpan RaceDuration { get; set; }
        public int TotalValue { get; set; }
        public int NumberOfPoints { get; set; }
        public int NumberOfStages { get; set; }
        public List<StageResultViewModel> StageResults { get; set; }
    }
}
